challenges-14-plus.yml

```diff
@@ -0,0 +1,472 @@
+              });
+          });
+          ```
+        c: |
+          ```javascript
+          document.addEventListener('turbo:visit', () => {
+              document.body.classList.add('turbo-loading');
+          });
+
+          document.addEventListener('turbo:before-render', (event) => {
+              event.detail.newBody.classList.add('turbo-loading');
+          });
+          ```
+        d: |
+          ```javascript
+          document.addEventListener('turbo:load', () => {
+              document.body.classList.add('turbo-loading');
+          });
+
+          document.addEventListener('turbo:before-render', (event) => {
+              event.detail.newBody.classList.add('turbo-loading');
+          });
+
+          document.addEventListener('turbo:render', () => {
+              requestAnimationFrame(() => {
+                 document.body.classList.remove('turbo-loading');
+              });
+          });
+          ```
+      correct: b
+  
+transitions_2:
+  question: |
+    Why is it hard in Turbo to implement page load transitions in versions 7.0.0 beta?
+    
+    > Notice that starting with version 7.0.0-rc.1 the "Pausable rendering" PR has been
+    > merged, making it easier to implement page transitions between clicks!
+  explanation: |
+    The fact that we cannot pause a render operation in Turbo before 7.0.0-rc.1 means that
+    our transition, if we make one, will always take the time it takes for the Ajax call
+    to finish. This makes it hard for us to perform complex transitions when we can't
+    *pause* the render until our animation properly finishes.
+  features:
+    multiple_chopice:
+      a: |
+        Because there are no good event handlers to use when transitioning pages for
+        this particular purpose.
+      b: |
+        Actually, you *can* easily implement page transitions between clicks in 7.0.0 beta.
+      c: |
+        Because wnen a page finishes loading, Turbo replaces the `<body>` element
+        immediately, leaving no time for a transition.
+      d: |
+        Because there are other libraries that can achieve this.
+    correct: c
+
+transition_finish:
+  question: |
+    Our lead designer is *ANGRY*! When our page went live, we've discovered that some
+    of our transitions didn't work as expected! Instead of fading to red and then back
+    to white, they faded from red to white twice!
+    
+    You can imagine the anger of our clients when they didn't get to properly experience
+    all those shades of pink.
+    
+    Can you tell me exactly when did this happen and why?
+  explanation: |
+    Our `turbo:before-render` and `turbo:render` events are flawed! They happen when a real
+    page, as well as a preview page, renders!
+    
+    This is the cause of the quirkiness that we've experienced. When the user travels to
+    a page that has a preview, our `turbo-loading` class is added and removed twice,
+    causing the double fade-in from red to white!
```

Comment:
I'm not sure we should keep this question. I think (?) we are basically asking "what is wrong with the implementation from 2 questions earlier?", right? It's really subtle and tricky.